rcgen/src/lib.rs

```diff
@@ -1141,17 +1153,28 @@ impl CertificateParams {
 	/// This key identifier is used in the SubjectKeyIdentifier X.509v3 extension.
 	fn key_identifier<K: PublicKeyData>(&self, pub_key: &K) -> Vec<u8> {
 		// Decide which method from RFC 7093 to use
+		#[cfg_attr(
+			not(feature = "crypto"),
+			allow(irrefutable_let_patterns, unused_variables)
```

Comment:
is allowing unused_variables still needed?

Reply:
Yes, it's needed because of this:

```
let digest_method = match &self.key_identifier_method {
```

Note that:
1. `irrefutable_let_patterns` is needed because only one match expression is used (`KeyIdMethod::PreSpecified(b)`)
2. `unused_variables` is needed because the following block is not used, so `digest_method` is unused.

This could be changed by refactoring perhaps so that the attributes aren't needed.